code/modules/assembly/flash.dm

```diff
@@ -322,6 +323,56 @@
 /obj/item/assembly/flash/cyborg/screwdriver_act(mob/living/user, obj/item/I)
 	return
 
+
+///Camera flash
+
+/obj/item/assembly/flash/camera
+	name = "camera"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	desc = "A polaroid camera."
+	icon_state = "camera"
+	inhand_icon_state = "camera"
+	worn_icon_state = "camera"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_NECK
+	pickup_sound = null
+	flash_sound = 'sound/items/polaroid1.ogg'
+	var/max_charges = 5
+	var/current_charges = 5
+	var/list/charge_timers = list()
+	var/charge_time = 300 //30 seconds
+
+/obj/item/assembly/flash/camera/burn_out()
+	return //use self charging system instead
+
+/obj/item/assembly/flash/camera/attack(mob/living/M, mob/user)
```

Comment:
Should this also lose a charge when it's used to flash an area? Currently it's only when you click someone with it.